### DISCUSSION TITLE
Can we talk about data validation?

### DISCUSSION THREAD

**Discussion by coltmay** [nss-day-cohort-47/C47-Discussions#55] [Help Me] [answered]
I'm really wanting to prevent adding a journal entry to my database accidentally.
Say I hit the **Save** button, but I didn't fill anything out in the forms.  Then it's just an object with undefined properties.

Could I use an if statement in the following code to somehow check if the variables are defined before posting??

```javascript
const clickSaveButton = () => {
    applicationElement.addEventListener("click", event => {
        event.preventDefault();
        if (event.target.id === "saveButton") {
            // A list of input variables is created and and assigned.
            const date = document.getElementById("journalDate").value;
            const concept = document.getElementById("journalConcepts").value;
            const mood = document.getElementById("journalMood").value;
            const entry = document.getElementById("journalEntry").value;

            // An object that takes in the above variables
            const entryObject = {
                date: date,
                concept: concept,
                mood: mood,
                entry: entry
            }
            // Posts object to the JSON file
            createEntry(entryObject);
        }
    })
}
```

**Comment by BryanNilsen** [comment, marked as answer]
**_- "Could I use an if statement in the following code to somehow check if the variables are defined before posting??"_**

YES!!! You can check to make sure some or all of the values are not empty strings, and then post the object. If any are empty, then maybe you can alert the user that they have to complete all of the fields...

**Reply by coltmay** [reply]
I didn't consider that they would be returned as empty strings, I thought they would be undefined, but we _do_ set that default value in the HTML, don't we?  Perfect, thanks!
